code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -573,10 +573,41 @@ GLOBAL_LIST_INIT(robot_modules, list(
 	modules += new /obj/item/extinguisher/mini(src) // For navigating space and/or low grav, and just being useful.
 	modules += new /obj/item/device/flash(src) // Non-lethal tool that prevents any 'borg from going lethal on Crew so long as it's an option according to laws.
 	modules += new /obj/item/crowbar/robotic(src) // Base crowbar that all 'borgs should have access to.
+	modules += new /obj/item/wetfloor_holder(src) // it hold da warning sign so hopefully no slippy :3
 	emag = new /obj/item/reagent_containers/spray(src)
 	emag.reagents.add_reagent(/singleton/reagent/lube, 250)
 	emag.name = "Lube spray"
 
+/obj/item/wetfloor_holder
+	name = "deployable wet floor sign"
+	desc = "A module that deploys a Wet Floor sign."
+	icon = 'icons/obj/janitor.dmi'
+	icon_state = "caution"
+	var/obj/item/clothing/suit/caution/held
+
+/obj/item/wetfloor_holder/New()
+	..()
+	held = new /obj/item/clothing/suit/caution(src)
+
+/obj/item/wetfloor_holder/attack_self(mob/user as mob)
+	if(!held)
+		to_chat(user, SPAN_NOTICE("The module is empty."))
+		return
+	var/obj/item/clothing/suit/caution/R = new /obj/item/clothing/suit/caution(user.loc)
+	to_chat(user, SPAN_NOTICE("You deploy \the [R]."))
+	R.add_fingerprint(user)
+	qdel(held)
+	held = null
+
+/obj/item/clothing/suit/caution/attackby(obj/item/attacking_item, mob/user)
+	if(istype(attacking_item, /obj/item/wetfloor_holder))
+		var/obj/item/wetfloor_holder/WFH = attacking_item
+		if(!WFH.held)
+			to_chat(user, SPAN_NOTICE("You collect \the [src]."))
+			forceMove(WFH)
+			WFH.held = src
+		return TRUE
```

Comment:
This whole block of code should be where the rest of the floor sign code is for consistency.